lib/modules/manager/poetry/schema.ts

```diff
@@ -132,58 +134,151 @@ export const PoetryDependencyRecord = LooseRecord(
     dep.skipReason = 'invalid-version';
     return dep;
   })
-).transform((record) => {
-  for (const [depName, dep] of Object.entries(record)) {
-    dep.depName = depName;
-    if (!dep.packageName) {
-      const pep503NormalizeRegex = regEx(/[-_.]+/g);
-      const packageName = depName
-        .toLowerCase()
-        .replace(pep503NormalizeRegex, '-');
-      if (depName !== packageName) {
-        dep.packageName = packageName;
+)
+  .transform((record) => {
+    const deps: PackageDependency[] = [];
+    for (const [depName, dep] of Object.entries(record)) {
+      dep.depName = depName;
+      if (!dep.packageName) {
+        const pep503NormalizeRegex = regEx(/[-_.]+/g);
+        const packageName = depName
+          .toLowerCase()
+          .replace(pep503NormalizeRegex, '-');
+        if (depName !== packageName) {
+          dep.packageName = packageName;
+        }
       }
+      deps.push(dep);
     }
-  }
-  return record;
-});
-
+    return deps;
+  })
+  .catch([]);
```

Comment:
why catch? what can go wrong? we should consider that whole file as invalid when we can't parse it